Tests/DependenciesMacrosTests/DependenciesMacrosTests.swift

```diff
@@ -48,20 +48,335 @@ final class DependenciesMacrosTests: XCTestCase {
+                func foo() -> String
+                func bar() -> String
+            }
+            class MyDependencyMock: MyDependencyProtocol {
+                var _foo: () -> String = unimplemented()
+                func foo() -> String {
+                    return _foo()
+                }
+                var _bar: () -> String = unimplemented()
+                func bar() -> String {
+                    return _bar()
+                }
+                init(foo: @escaping () -> String = unimplemented(), bar: @escaping () -> String = unimplemented()) {
+                    self._foo = foo
+                    self._bar = bar
+                }
             }
-            @DumpSyntax
-            public class Henk {
-                var implicit = Date(timeIntervalSinceReferenceDate: 200)
+            extension MyDependency: MyDependencyProtocol {
             }
             """,
-            expandedSource: "",
-            macros: ["DumpSyntax": DumpSyntaxMacro.self]
+            macros: DependenciesMacrosPlugin.macros
+        )
+    }
+
+    func testFunctionWithParameter() {
+        assertMacroExpansion(
+            """
+            @AutoDependency
+            final class MyDependency {
+                func foo(bar: String) -> String {
+                    return bar
+                }
+            }
+            """,
+            expandedSource: """
+            final class MyDependency {
+                func foo(bar: String) -> String {
+                    return bar
+                }
+            }
+            protocol MyDependencyProtocol : AnyObject {
+                func foo(bar: String) -> String
+            }
+            class MyDependencyMock: MyDependencyProtocol {
+                var _foo: (String) -> String = unimplemented()
+                func foo(bar: String) -> String {
+                    return _foo(bar)
+                }
+                init(foo: @escaping (String) -> String = unimplemented()) {
+                    self._foo = foo
+                }
+            }
+            extension MyDependency: MyDependencyProtocol {
+            }
+            """,
+            macros: DependenciesMacrosPlugin.macros
+        )
+    }
+
+    func testFunctionWithMultipleParameters() {
+        assertMacroExpansion(
+            """
+            @AutoDependency
+            final class MyDependency {
+                func foo(bar: String, baz: Int) -> String {
+                    return bar
+                }
+            }
+            """,
+            expandedSource: """
+            final class MyDependency {
+                func foo(bar: String, baz: Int) -> String {
+                    return bar
+                }
+            }
+            protocol MyDependencyProtocol : AnyObject {
+                func foo(bar: String, baz: Int) -> String
+            }
+            class MyDependencyMock: MyDependencyProtocol {
+                var _foo: (String, Int) -> String = unimplemented()
```

Comment:
Just wondering, but what's the use of this default value i the `init` already overrides it with a default?

Reply:
Absolutely nothing at all 😉